homeassistant/components/acmeda/cover.py

```diff
@@ -0,0 +1,60 @@
+"""Support for Acmeda Roller Blinds."""
+import asyncio
+
+from homeassistant.components.cover import ATTR_POSITION, CoverDevice
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .base import AcmedaBase
+from .const import ACMEDA_HUB_UPDATE
+from .helpers import update_entities
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Acmeda Rollers from a config entry."""
+    update_lock = asyncio.Lock()
+    current = {}
+
+    async def async_update():
+        async with update_lock:
+            await update_entities(
+                hass, AcmedaCover, config_entry, current, async_add_entities
+            )
+
+    async_dispatcher_connect(
```

Comment:
We should unsubscribe the dispatcher in `async_unload_entry` in the base integration module. Store the return value here in hass.data, and use the returned function to unsubscribe.

Reply:
Added a cleanup_callback list to the PulseHub class.
The returned functions are called on async_reset of the PulseHub object.